tests/test_session_scope.py

```diff
@@ -71,3 +72,26 @@ def test_generic_scope(database):
         malbolge = session.query(database.Language).first()
         assert malbolge.name == 'Malbolge'
         assert malbolge.created == 1998
+
+
+@pytest.mark.parametrize('cleanup,expected', [
+    (SessionCleanup.CLOSE_ONLY, 1999),
+    (SessionCleanup.COMMIT, 1998),
+    (SessionCleanup.COMMIT_ON_SUCCESS, 1998),
```

Comment:
we are not testing that it does not commit on error

Reply:
I need to rework the tests of the package wholesale, but I've now added a test for that.
Not sure if it makes sense to have that "commit" mode that tries to commit regardless?